deployment/tfhub/object-detection/base64/object-detection-base64-predict.ipynb

```diff
@@ -0,0 +1,241 @@
+    "    urls = [\n",
+    "        'http://s3.amazonaws.com/verta-starter/street-view-images/000001_0.jpg',\n",
+    "        'http://s3.amazonaws.com/verta-starter/street-view-images/000001_1.jpg',\n",
+    "        'http://s3.amazonaws.com/verta-starter/street-view-images/000001_2.jpg',\n",
+    "        'http://s3.amazonaws.com/verta-starter/street-view-images/000001_3.jpg',\n",
+    "        'http://s3.amazonaws.com/verta-starter/street-view-images/000001_4.jpg'\n",
+    "    ]\n",
+    "    urls = urls[:n_imgs]\n",
+    "    data = []\n",
+    "\n",
+    "    for url in urls:\n",
+    "        file = url.split('/')[-1]\n",
+    "        wget.download(url, file)\n",
+    "        \n",
+    "        with open(file, 'rb') as img:\n",
+    "            img_bytes = base64.b64encode(img.read())\n",
+    "            img_str = img_bytes.decode('utf-8')\n",
+    "            img_str = json.dumps(img_str)\n",
+    "        \n",
+    "        data.append([file, img_str])\n",
+    "\n",
+    "    return data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def show_metrics(n_imgs, n_threads, start_time, end_time):\n",
+    "    total_time = end_time - start_time\n",
+    "    total_time = time.strftime('%Mm %Ss', time.gmtime(total_time))\n",
+    "    \n",
+    "    print(f'Images processed: {n_imgs}.')\n",
+    "    print(f'Threads: {n_threads}.')\n",
+    "    print(f'Total time: {total_time}.')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def show_results(results):\n",
+    "    cols = list(results[0].keys())[:-1]\n",
+    "    cols.extend(list(results[0]['bboxes'].keys()))\n",
+    "    data = []\n",
+    "\n",
+    "    for item in results:\n",
+    "        values = list(item.values())[0:3]\n",
+    "        values.extend(list(item['bboxes'].values()))\n",
+    "        data.append(values)\n",
+    "\n",
+    "    df = pd.DataFrame(data, columns=cols)\n",
+    "    print(df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Main Function\n",
+    "After setting up Verta, it processes the images in parallel, presents metrics and saves the result to a CSV file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def main():\n",
+    "    VERTA_HOST = 'app.verta.ai'\n",
+    "    ENDPOINT_NAME = 'object-detection-base64'\n",
+    "\n",
+    "    os.environ['VERTA_EMAIL'] = ''\n",
+    "    os.environ['VERTA_DEV_KEY'] = ''\n",
+    "\n",
+    "    client = Client(VERTA_HOST, debug=True)\n",
+    "    endpoint = client.get_or_create_endpoint(ENDPOINT_NAME)\n",
+    "    model = endpoint.get_deployed_model()\n",
+    "\n",
+    "    n_threads = int(mp.cpu_count())\n",
+    "    n_imgs = 1\n",
+    "    imgs = load_imgs(n_imgs)\n",
```

Comment:
It looks like this only uses img #1, which is strange to do in parallel. If we have enough RAM, it might make more sense to load every image?

Reply:
Sure, that's with only 1image just for testing purpose. It works with all the 10 images available, but @mvartakAtVerta warned me about an OOM error when running with 10k images, and I'll work on that in a different PR :+1: